server/index.js

```diff
@@ -0,0 +1,205 @@
+require('dotenv').config();
+const request = require('request');
+const http = require('http');
+const httpProxy = require('http-proxy');
+const express = require('express');
+const bodyParser = require('body-parser');
+const bcrypt = require('bcrypt');
+const JWT = require('./jwt');
+
+const app = express();
+
+const proxy = httpProxy.createProxyServer({
+  changeOrigin: true,
+  ignorePath: false
+});
+
+proxy.on('proxyReq', function(proxyReq, req, res, options) {
+  proxyReq.setHeader('authorization', `Bearer ${process.env.AIRTABLE_API_KEY}`);
+});
+
+app.use(bodyParser.json());
+
+app.get('/v0*', JWT.verifyToken, (req, res) => {
+  proxy.web(req, res, {
+    target: `${process.env.AIRTABLE_ENDPOINT_URL}`
+  });
+});
+
+app.post('/v0*', JWT.verifyToken, (req, res) => {
+  proxy.web(req, res, {
+    target: `${process.env.AIRTABLE_ENDPOINT_URL}`
+  });
+});
+
+app.put('/v0*', JWT.verifyToken, (req, res) => {
+  proxy.web(req, res, {
+    target: `${process.env.AIRTABLE_ENDPOINT_URL}`
+  });
+});
+
+app.delete('/v0*', JWT.verifyToken, (req, res) => {
+  proxy.web(req, res, {
+    target: `${process.env.AIRTABLE_ENDPOINT_URL}`
+  });
+});
+
+app.post('/register', (req, res) => {
+  const username = req.body.username;
+  const password = req.body.password;
+
+  const url = `${process.env.AIRTABLE_ENDPOINT_URL}/v0/${process.env.AIRTABLE_BASE_ID}/${process.env.AIRTABLE_USER_TABLE}?fields%5B%5D=username&filterByFormula=username%3D%22${username}%22`;
+  const headers = {
+    authorization: 'Bearer ' + process.env.AIRTABLE_API_KEY,
+    'x-api-version': process.env.AIRTABLE_API_VERSION,
+    'x-airtable-application-id': process.env.AIRTABLE_BASE_ID,
+    'User-Agent': process.env.AIRTABLE_USER_AGENT
+  };
+  const options = {
+    method: 'GET',
+    url: url,
+    json: true,
+    timeout: 5000,
+    headers: headers,
+    agentOptions: {
+      rejectUnauthorized: false
+    }
+  };
+  const urlCreate = `${process.env.AIRTABLE_ENDPOINT_URL}/v0/${process.env.AIRTABLE_BASE_ID}/${process.env.AIRTABLE_USER_TABLE}?`;
+  var optionsCreate = {
+    method: 'POST',
+    url: urlCreate,
+    json: true,
+    timeout: 5000,
+    headers: headers,
+    agentOptions: {
+      rejectUnauthorized: false
+    },
+    body: {
+      fields: {
+        username: `${username}`,
+        password: `${password}`
+      }
+    }
+  };
+  request(options, async function(error, resp, body) {
+    if (error) {
+      res.send(error);
+    }
+    if (body.records.length > 0) {
+      const payload = { success: false, message: 'user exists' };
+      res.status(422).send(payload);
+      res.send(payload);
+    } else {
+      if (JSON.parse(process.env.HASH_PASSWORD) === true) {
+        bcrypt.hash(
+          req.body.password.trim(),
```

Comment:
Why do we want to `.trim()` the password here?

Reply:
Remove any accidental space characters introduced before or after the password. 

Use case: User accidently types a space before or after typing their password.

But I also think that if we're not offering any password format validation beyond a .trim() I can remove the .trim() and assume the client will be handling all password format validation on the client side.

The other way I was thinking -- which would take more effort is to allow a developer to define a password validator function that they could inject. 

But we could just remove the .trim() if that makes it more clear as to who (client vs airlock) is in charge of validating/cleaning up the password format

Reply:
I think we should just remove the `.trim`. This being primarily an authentication/API library, I don't want to introduce unnecessary responsibility on our end where the developer might not expect it.

Reply:
Perfect! Commit on 10/21/19 reflects the removal of .trim()